sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
 | Key Vault ([example][kv_tenant_id]) | Request access in a tenant that was discovered as part of an authentication challenge
 
 
+## Long-running operation (LRO) customization
+
+ API operations that can take longer to complete are often represented as long-running operations (LRO). `azure-core` provides `LROPoller` (and `AsyncLROPoller`) protocols
+ that give users access to the LRO and expose methods to interact with them such as waiting until the operation reaches a terminal state, checking its status, or
+ providing a callback to do work on the final result when it is ready. If the LRO follows the [Azure REST API guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations--jobs), 
+ it's likely that the generated client library code should _just work_. 
+ In cases where the LRO diverges from the guidelines, you may need to customize your code to achieve the desired scenario.
+
+ There are 3 options to customize the logic for our LROs.
+
+ 1) [LRO algorithm/strategies - OperationResourcePolling, LocationPolling, StatusCheckPolling](#lro-algorithmstrategies---operationresourcepolling-locationpolling-statuscheckpolling)
+    - You need to customize the polling strategy
+ 2) [Polling method - LROBasePolling/AsyncLROBasePolling](#polling-method---lrobasepollingasynclrobasepolling)
+    - You need to customize the polling loop
+ 3) [Poller - LROPoller/AsyncLROPoller](#poller---lropollerasynclropoller)
+    - You need to customize the public interface of the poller
+
+
+### LRO algorithm/strategies - OperationResourcePolling, LocationPolling, StatusCheckPolling
+
+`azure.core.polling` provides three built-in strategies for polling - `OperationResourcePolling`, `LocationPolling`, and `StatusCheckPolling`.
+The type of polling needed will be determined automatically, unless otherwise changed by the client library developer. If the LRO is determined
+not to fit `OperationResourcePolling` or `LocationPolling`, `StatusCheckPolling` serves as a fallback strategy which will not perform polling, but
+instead return a successful response for a 2xx.
+
+If you need to customize the polling strategy, choose a polling algorithm that closely represents what you need to do
+or create your own that inherits from `azure.core.polling.LongRunningOperation` and implements the necessary methods.
```

Comment:
Ref docs? Would be helpful to be able to click through and see what those "necessary methods" might be.